lib/core/test/integration/Test/Integration/Scenario/CLI/Server.hs

```diff
@@ -0,0 +1,63 @@
+module Test.Integration.Scenario.CLI.Server
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Launcher
+    ( Command (..), StdStream (..))
+import Test.Hspec.Expectations.Lifted
+    ( shouldBe )
+import Control.Monad
+    ( when )
+import System.Directory
+    ( doesFileExist, removePathForcibly )
+import Test.Hspec
+    ( SpecWith, after_, describe, it )
+import Test.Integration.Framework.DSL
+    ( Context (..), expectCmdStarts )
+
+spec :: SpecWith (Context t)
+spec = after_ tearDown $ do
+    describe "SERVER - cardano-wallet server" $ do
+        it "SERVER - Can start cardano-wallet server without --database" $ \_ -> do
+            let cardanoWalletServer = Command
+                    "cardano-wallet"
+                    [ "server"
+                    , "--random-port"
```

Comment:
I'd remove the `--random-port` from the tests since it's completely unrelated and this is a totally optional argument.